estate/models/estate_property.py

```diff
@@ -0,0 +1,101 @@
+from dateutil.relativedelta import relativedelta
+from odoo import fields, models, api
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.float_utils import float_is_zero, float_compare
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "A property managed by the Estate module."
+
+    name = fields.Char(required=True)
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date(copy=False, default= fields.Date.today()+relativedelta(months=3))
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float(readonly=True, copy=False, default=0)
+
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer()
+    facades = fields.Integer()
+
+    active = fields.Boolean(default=True)
+    state = fields.Char()
+
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area= fields.Integer()
+    garden_orientation = fields.Selection(
+        selection=[("north", "North"), ("south", "South"), ("east", "East"), ("west", "West")]
+    )
+    state = fields.Selection(
+        selection=[("new", "New"), ("offer_received", "Offer Received"), ("offer_accepted", "Offer Accepted"), ("sold", "Sold"), ("cancelled", "Cancelled") ],
+        copy=False,
+        default="new",
+        required=True
+    )
+
+    salesman_id = fields.Many2one("res.users", string="Salesman", default=lambda self: self.env.user)
+    buyer_id = fields.Many2one("res.partner", string="Buyer", copy=False)
+    property_type_id = fields.Many2one("estate.property.type")
+    property_tag_ids = fields.Many2many("estate.property.tag")
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offers")
+
+    total_area = fields.Integer(compute="_compute_total_area")
+
+    best_price = fields.Float(compute="_compute_best_price")
+
+    # Do SQL constraints need to be written with single-quotes, as you would in an actual SQL expression? Or are double-quotes always ok?
+    _sql_constraints = [
+        ("expected_price", "CHECK(expected_price > 0)", "Expected price must be positive" ),
+        ("selling_price", "CHECK(selling_price >= 0)", "Can't sell for a negative price!")
+    ]
+
+    @api.constrains("selling_price", "expected_price")
+    def _price_in_expectations(self):
+        for record in self:
+            if not float_is_zero(record.selling_price, precision_digits=2) and float_compare(record.selling_price, record.expected_price * 0.9, precision_digits=2) < 0:
+                raise ValidationError("You may not accept an offer for a price inferior to 90%% of expected sale price!")
+
+
+    @api.depends("garden_area", "living_area")
+    def _compute_total_area(self):
+        for record in self:
+            lr = record.living_area if record.living_area else 0
+            gr = record.garden_area if record.garden_area else 0
```

Comment:
Two alternatives if you want to use shorthands
```suggestion
            lr = record.living_area or 0
            gr = int(record.garden_area)
```